### PR TITLE
Feature/frontend

### PR DESCRIPTION
**Pull request by rajpatil53**

### CODE REVIEW

File: contracts/SeedsOnEarth.sol [hunk 1]
```diff
@@ -83,23 +85,36 @@ contract SeedsOnEarth {
...
+    /**
+    * @notice join a quest, called by user to join a quest
+    * @param _questId id of quest
+    **/
+    function joinQuest(uint256 _questId) public {
+        Quest storage quest = quests[_questId];
+        require(quest.status == QuestStatus.PENDING, "Quest must be pending to join");
+        quest.users.push(msg.sender);
+        usersQuestsMapping[msg.sender].push(_questId);
+        if (quest.users.length == quest.numOfUsers) {
```
Comment (lirona) [inline contracts/SeedsOnEarth.sol:100]: this will always be true

**Review by lirona — COMMENTED** [review]
lgtm, except for the array thing

File: contracts/SeedsOnEarth.sol (hunk 1 above)
Comment (rajpatil53) [reply contracts/SeedsOnEarth.sol:100]: Understood, Liron.

Assigning dynamic size array throws this error:
```
Invalid type for argument in function call. Invalid implicit conversion from function (uint256) pure returns (address[] memory) to address[] memory requested.
```

Should I create a separate counter for number of users?

File: contracts/SeedsOnEarth.sol [hunk 2]
```diff
@@ -150,7 +165,7 @@ contract SeedsOnEarth {
              "Quest can only be reset if it was dismissed by sponser or time to complete had passed");
         quest.pickedUpHash = "";
         quest.completedHash = "";
-        quest.users = new address[];
+        quest.users = new address[](quest.numOfUsers);
```
Comment (lirona) [inline contracts/SeedsOnEarth.sol]: should be (0) again

File: contracts/SeedsOnEarth.sol (hunk 1 above)
Comment (lirona) [reply contracts/SeedsOnEarth.sol:100]: only problem is that now if the quest requires only one user, the user will have to "join" and then "pickup"

Comment (lirona) [reply contracts/SeedsOnEarth.sol:100]: and then can keep the quest stuck with just joining.. but that's ok for now i guess

Comment (rajpatil53) [reply contracts/SeedsOnEarth.sol:100]: That's right.
I think I can handle that in frontend for now. If `numOfUsers` is 1, then I will run `pickupQuest` right after `joinQuest`

**Pull request merged**